packages/app-desktop/commands/openNoteInNewWindow.ts

```diff
@@ -0,0 +1,35 @@
+import { CommandRuntime, CommandDeclaration, CommandContext } from '@joplin/lib/services/CommandService';
+import { _ } from '@joplin/lib/locale';
+import { stateUtils } from '@joplin/lib/reducer';
+import Note from '@joplin/lib/models/Note';
+import { createAppDefaultWindowState } from '../app.reducer';
+import Setting from '@joplin/lib/models/Setting';
+
+export const declaration: CommandDeclaration = {
+	name: 'openNoteInNewWindow',
+	label: () => _('Edit in new window'),
+	iconName: 'icon-share',
+};
+
+let idCounter = 0;
+
+export const runtime = (): CommandRuntime => {
+	return {
+		execute: async (context: CommandContext, noteId: string = null) => {
+			noteId = noteId || stateUtils.selectedNoteId(context.state);
+			const note = await Note.load(noteId, { fields: ['parent_id'] });
```

Comment:
I think noteId could be undefined? If there's no note in the notebook for example

Reply:
The `enabledCondition` is currently `oneNoteSelected`. As such, the command should only run if triggered by a script or `oneNoteSelected` is true. To make this requirement clearer, I've added a `throw new Error` if `!noteId`.

Reply:
No you're right, with the condition the note should indeed be defined so the exception is not necessary. Could you remove it please?